 Extracción de Datos a Nivel Nacional

In [1]:
import requests

In [2]:
import pandas as pd

In [16]:
pd.set_option('display.max_columns', None)  # para poder visualizar todas las columnas de los DataFrames

In [3]:
url = "https://apidatos.ree.es/es/datos/generacion/estructura-generacion?start_date=2011-01-01T00:00&end_date=2011-01-31T23:59&time_trunc=month"

energia = requests.get(url)


In [4]:
energia.status_code

200

In [5]:
generacion_energia = energia.json()

In [7]:
len(generacion_energia)

2

In [9]:
generacion_energia.keys()


dict_keys(['data', 'included'])

In [10]:
generacion_energia["data"]

{'type': 'Generación por tecnología',
 'id': 'gen1',
 'attributes': {'title': 'Generación por tecnología',
  'last-update': '2019-06-12T15:32:24.000+02:00',
  'description': None},
 'meta': {'cache-control': {'cache': 'MISS'}}}

In [ ]:
generacion_energia["included"]

In [13]:
type(generacion_energia["included"])

list

In [14]:
generacion_energia["included"][0]

{'type': 'Hidráulica',
 'id': '10288',
 'groupId': '1',
 'attributes': {'title': 'Hidráulica',
  'description': None,
  'color': '#0090d1',
  'type': 'Renovable',
  'magnitude': None,
  'composite': False,
  'last-update': '2019-06-12T15:32:24.000+02:00',
  'values': [{'value': 5119512.38,
    'percentage': 0.19757364151478785,
    'datetime': '2011-01-01T00:00:00.000+01:00'}]}}

In [17]:
df_include = pd.DataFrame(generacion_energia["included"])
df_include.head()


type     id groupId  \
0          Hidráulica  10288       1   
1  Turbinación bombeo   1445       1   
2             Nuclear   1446       1   
3              Carbón  10289       1   
4          Fuel + Gas  10290       1   

                                          attributes  
0  {'title': 'Hidráulica', 'description': None, '...  
1  {'title': 'Turbinación bombeo', 'description':...  
2  {'title': 'Nuclear', 'description': None, 'col...  
3  {'title': 'Carbón', 'description': None, 'colo...  
4  {'title': 'Fuel + Gas', 'description': None, '...

In [18]:
generacion_energia["included"][0]['attributes']

{'title': 'Hidráulica',
 'description': None,
 'color': '#0090d1',
 'type': 'Renovable',
 'magnitude': None,
 'composite': False,
 'last-update': '2019-06-12T15:32:24.000+02:00',
 'values': [{'value': 5119512.38,
   'percentage': 0.19757364151478785,
   'datetime': '2011-01-01T00:00:00.000+01:00'}]}

In [19]:
df_include = pd.DataFrame(generacion_energia["included"][0]['attributes'])
df_include.head()

title description    color       type magnitude  composite  \
0  Hidráulica        None  #0090d1  Renovable      None      False   

                     last-update  \
0  2019-06-12T15:32:24.000+02:00   

                                              values  
0  {'value': 5119512.38, 'percentage': 0.19757364...

'title': 'Hidráulica'
'type': 'Renovable'
'last-update': '2019-06-12T15:32:24.000+02:00'
'values': [{'value': 5119512.38,
   'percentage': 0.19757364151478785,
   'datetime': '2011-01-01T00:00:00.000+01:00'}]

In [24]:
dicc_generacion = {"tipo_energia":[], "tipo_generacion":[], "fecha":[], "valor":[]}

for gen in generacion_energia["included"]:
    dicc_generacion ["tipo_energia"].append(gen["attributes"]['title'])
    dicc_generacion ["tipo_generacion"].append(gen["attributes"]['type'])
    dicc_generacion ["fecha"].append(gen["attributes"]['last-update'])
    dicc_generacion ["valor"].append(gen["attributes"]['values'][0]['value'])
    


In [ ]:
dicc_generacion

In [27]:
df_generacion = pd.DataFrame(dicc_generacion)
df_generacion.head()

tipo_energia tipo_generacion                          fecha  \
0          Hidráulica       Renovable  2019-06-12T15:32:24.000+02:00   
1  Turbinación bombeo    No-Renovable  2019-06-12T15:32:24.000+02:00   
2             Nuclear    No-Renovable  2018-12-10T12:55:09.000+01:00   
3              Carbón    No-Renovable  2018-12-21T10:43:14.000+01:00   
4          Fuel + Gas    No-Renovable  2018-12-10T12:55:14.000+01:00   

         valor  
0  5119512.380  
1   283944.912  
2  4706500.878  
3  2303087.707  
4     -751.170